fe/fe-core/src/main/java/org/apache/doris/datasource/EsExternalDataSource.java

```diff
@@ -166,7 +167,10 @@ public List<String> listDatabaseNames(SessionContext ctx) {
 
     @Override
     public List<String> listTableNames(SessionContext ctx, String dbName) {
-        return esRestClient.getIndexes();
+        List<String> indexes = esRestClient.getIndexes().stream().distinct().collect(Collectors.toList());
```

Comment:
I think maybe package this line into esRestClient is better, provide `listTable` method

Reply:
Good idea. Let me modify it.